src/Castle.Core/DynamicProxy/IChangeProxyTarget.cs

```diff
@@ -53,6 +53,7 @@ public interface IChangeProxyTarget
 		///    cref = "InvalidOperationException" /> will be throws.
 		/// </remarks>
 		/// <exception cref = "InvalidCastException">Thrown when <paramref name = "target" /> is not assignable to the proxied type.</exception>
+		[Obsolete("Use (invocation.Proxy as IProxyTargetAccessor).DynProxySetTarget(target) instead.")]
```

Comment:
I'm going to be picky here since this is in the actual code, it should really be a normal cast rather than a safe cast, an invalid cast exception is better than a null ref exception.